### ISSUE TITLE
[Bug maybe?] A variable could be overwritten immediately

### ISSUE THREAD

**Issue by LEON-REIN** [Sakib1263/ResNet-ResNetV2-SEResNet-ResNeXt-SEResNeXt-1D-2D-Tensorflow-Keras#4]
## Bug location in the codes
There have been many cases like this and I'll show one location:
> SE_ResNet_1DCNN.py -->  SEResNet --> MLP (line 232~236 )

## My personal understanding
If `self.problem_type == 'Classification'`, then the variable `outputs` would be covered by the following softmax layer. This causes unnecessary calculation overhead.

Thanks for your contribution, sincerely!

**Comment by LEON-REIN** [comment]
Oh, I forgot my solution:
```python
return tf.keras.layers.Dense(self.output_nums, activation='softmax')(x) \
    if self.problem_type == 'Classification' \
    else tf.keras.layers.Dense(self.output_nums, activation='linear')(x)
```

**Comment by Sakib1263** [comment]
It's not a bug, just a different way to perform the same thing. Computational Efficiency might be different.
